src/components/Proposal/ProposalActions.tsx

```diff
@@ -1,30 +1,28 @@
 import React from 'react'
 
+import useAuthContext from 'decentraland-gatsby/dist/context/Auth/useAuthContext'
 import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
 import { Button } from 'decentraland-ui/dist/components/Button/Button'
 
-import { ProposalAttributes } from '../../entities/Proposal/types'
-import { ProposalStatus } from '../../entities/Proposal/types'
+import { ProposalAttributes, ProposalStatus } from '../../entities/Proposal/types'
+import useIsCommittee from '../../hooks/useIsCommittee'
+import useIsProposalOwner from '../../hooks/useIsProposalOwner'
 import { ProposalPageState } from '../../pages/proposal'
 
 type ProposalActionsProps = {
-  isOwner: boolean
-  isCommittee: boolean
   deleting: boolean
   updatingStatus: boolean
   proposal: ProposalAttributes | null
   updatePageState: (newState: Partial<ProposalPageState>) => void
 }
 
-export default function ProposalActions({
-  proposal,
-  isOwner,
-  isCommittee,
-  deleting,
-  updatingStatus,
-  updatePageState,
-}: ProposalActionsProps) {
+export default function ProposalActions({ proposal, deleting, updatingStatus, updatePageState }: ProposalActionsProps) {
   const t = useFormatMessage()
+  const [account] = useAuthContext()
+  const { isCommittee } = useIsCommittee(account)
+  const { isOwner } = useIsProposalOwner(proposal)
+
+  if (!proposal) return null
```

Comment:
You can move this condition to wherever you are using this component
`{proposal && <ProposalActions {...props} />`

Reply:
Thanks :), I purposely tried to avoid that. 

My usual approach is: if the component needs to load a lot of data before returning, then try to decide if rendering is necessary from outside the component (to avoid having those expensive calls)

If the component is `lightweight` and does not need much to return, then make the decision inside it so it can own the responsibility, and hide the complexity. (And in any case we need to reuse it, we don't need to rewrite the condition)

WDYT?